samza-core/src/test/java/org/apache/samza/container/grouper/stream/TestAllSspToSingleTaskGrouper.java

```diff
@@ -0,0 +1,108 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.samza.Partition;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigException;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.container.TaskName;
+import org.apache.samza.system.SystemStreamPartition;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+public class TestAllSspToSingleTaskGrouper {
+  private SystemStreamPartition aa0 = new SystemStreamPartition("SystemA", "StreamA", new Partition(0));
+  private SystemStreamPartition aa1 = new SystemStreamPartition("SystemA", "StreamA", new Partition(1));
+  private SystemStreamPartition aa2 = new SystemStreamPartition("SystemA", "StreamA", new Partition(2));
+  private SystemStreamPartition ab0 = new SystemStreamPartition("SystemA", "StreamB", new Partition(0));
+  private AllSspToSingleTaskGrouperFactory grouperFactory = new AllSspToSingleTaskGrouperFactory();
+
+  @Test
+  public void testLocalStreamGroupedCorrectlyForYarn() {
+    HashSet<SystemStreamPartition> allSSPs = new HashSet<>();
+    HashMap<String, String> configMap = new HashMap<>();
+
+    configMap.put("yarn.container.count", "2");
+
+    Config config = new MapConfig(configMap);
+
+    SystemStreamPartitionGrouper grouper = grouperFactory.getSystemStreamPartitionGrouper(config);
+
+    Collections.addAll(allSSPs, aa0, aa1, aa2, ab0);
+    Map<TaskName, Set<SystemStreamPartition>> result = grouper.group(allSSPs);
+    Map<TaskName, Set<SystemStreamPartition>> expectedResult = new HashMap<>();
+
+    HashSet<SystemStreamPartition> partitions = new HashSet<>();
+    partitions.add(aa0);
+    partitions.add(aa1);
+    partitions.add(aa2);
+    partitions.add(ab0);
+    expectedResult.put(new TaskName("Task-0"), partitions);
+    expectedResult.put(new TaskName("Task-1"), partitions);
+
+    assertEquals(expectedResult, result);
+  }
+
+  @Test
+  public void testLocalStreamGroupedCorrectlyForPassthru() {
+    HashSet<SystemStreamPartition> allSSPs = new HashSet<>();
+    HashMap<String, String> configMap = new HashMap<>();
+
+    configMap.put("processor.id", "1");
+    configMap.put("job.coordinator.factory", "org.apache.samza.standalone.PassthroughJobCoordinatorFactory");
+
+    Config config = new MapConfig(configMap);
+
+    SystemStreamPartitionGrouper grouper = grouperFactory.getSystemStreamPartitionGrouper(config);
+
+    Collections.addAll(allSSPs, aa0, aa1, aa2, ab0);
+    Map<TaskName, Set<SystemStreamPartition>> result = grouper.group(allSSPs);
+    Map<TaskName, Set<SystemStreamPartition>> expectedResult = new HashMap<>();
+
+    HashSet<SystemStreamPartition> partitions = new HashSet<>();
+    partitions.add(aa0);
+    partitions.add(aa1);
+    partitions.add(aa2);
+    partitions.add(ab0);
+    expectedResult.put(new TaskName("Task-1"), partitions);
+
+    assertEquals(expectedResult, result);
+  }
+
+  @Test(expected = ConfigException.class)
```

Comment:
How about other test cases? Like PassthroughJobCoordinator failed to get processorId, ZkJobCoordinator failed to get the processor count, etc.?

Reply:
PassthroughJobCoordinator should have processorId even if it is not explicitly added in the config due to the default processorId generator. Added a test for it.

As described above, I will talk to you about the UT for ZkJobCoordinator.